File: samples/Demo/Data/DemoSeeder.cs
namespace Demo.Data;

public static class DemoSeeder
{
    public static void SeedIfEmpty(AppDb db)
    {
        if (db.Customers.Any()) return;

        var acme = new Customer { Name = "Acme Corp", Email = "buyer@acme.test" };
        var globex = new Customer { Name = "Globex", Email = "ops@globex.test" };
        var initech = new Customer { Name = "Initech", Email = "ap@initech.test" };

        db.Customers.AddRange(acme, globex, initech);

        var statuses = new[]
        {
            OrderStatus.Pending, OrderStatus.Processing, OrderStatus.Shipped,
            OrderStatus.Delivered, OrderStatus.Cancelled,
        };
        var customers = new[] { acme, globex, initech };

        var orders = Enumerable.Range(1, 42).Select(i => new Order
        {
            Reference = $"ORD-{1000 + i}",
            Customer = customers[i % customers.Length],
            Status = statuses[i % statuses.Length],
            Total = 19.99m + i * 7.25m,
            CreatedAt = DateTime.UtcNow.AddDays(-i),
        });

        db.Orders.AddRange(orders);
        db.SaveChanges();
    }
}

File: samples/FilamentReference/database/seeders/DemoSeeder.php
<?php

namespace Database\Seeders;

use App\Enums\OrderStatus;
use App\Models\Customer;
use App\Models\Order;
use Illuminate\Database\Seeder;

class DemoSeeder extends Seeder
{
    /**
     * Seed the application's database.
     */
    public function run(): void
    {
        if (Customer::withTrashed()->exists()) {
            return;
        }

        $acme = Customer::create(['name' => 'Acme Corp', 'email' => 'buyer@acme.test']);
        $globex = Customer::create(['name' => 'Globex', 'email' => 'ops@globex.test']);
        $initech = Customer::create(['name' => 'Initech', 'email' => 'ap@initech.test']);

        $statuses = [
            OrderStatus::Pending,
            OrderStatus::Processing,
            OrderStatus::Shipped,
            OrderStatus::Delivered,
            OrderStatus::Cancelled,
        ];
        $customers = [$acme, $globex, $initech];

        for ($i = 1; $i <= 42; $i++) {
            Order::create([
                'reference' => 'ORD-'.(1000 + $i),
                'customer_id' => $customers[$i % count($customers)]->id,
                'status' => $statuses[$i % count($statuses)],
                'total' => 19.99 + $i * 7.25,
                'created_at' => now()->subDays($i),
            ]);
        }
    }
}
